Server/DeviceManagement/DeviceManagement.Application/CQRS/CommandHandlers/CreateRouterCommandHandler.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using ModularHouse.Libraries.InternalMessaging.CQRS.Abstractions.Command;
+using ModularHouse.Libraries.InternalMessaging.DomainEvents.Abstractions;
+using ModularHouse.Server.Common.Domain;
+using ModularHouse.Server.Common.Domain.Exceptions;
+using ModularHouse.Server.DeviceManagement.Application.CQRS.Commands;
+using ModularHouse.Server.DeviceManagement.Domain.AreaAggregate;
+using ModularHouse.Server.DeviceManagement.Domain.DeviceAggregate;
+using ModularHouse.Server.DeviceManagement.Domain.RouterAggregate;
+using ModularHouse.Server.DeviceManagement.Domain.RouterAggregate.Events;
+using ModularHouse.Server.DeviceManagement.Domain.UserAggregate;
+
+namespace ModularHouse.Server.DeviceManagement.Application.CQRS.CommandHandlers;
+
+public class CreateRouterCommandHandler : ICommandHandler<CreateRouterCommand>
+{
+    private readonly IRouterRepository _routerRepository;
+    private readonly IUserDataSource _userDataSource;
+    private readonly IAreaDataSource _areaDataSource;
+    private readonly IDeviceDataSource _deviceDataSource;
+    private readonly IRouterDataSource _routerDataSource;
+    private readonly IDomainEventBus _eventBus;
+
+    public CreateRouterCommandHandler(
+        IRouterRepository routerRepository,
+        IUserDataSource userDataSource,
+        IAreaDataSource areaDataSource, 
+        IDeviceDataSource deviceDataSource,
+        IRouterDataSource routerDataSource,
+        IDomainEventBus eventBus)
+    {
+        _routerRepository = routerRepository;
+        _userDataSource = userDataSource;
+        _areaDataSource = areaDataSource;
+        _deviceDataSource = deviceDataSource;
+        _routerDataSource = routerDataSource;
+        _eventBus = eventBus;
+    }
+
+    public async Task Handle(CreateRouterCommand command, CancellationToken cancellationToken)
+    {
+        await ThrowIfUserIsNotExistByIdAsync(command.UserId, cancellationToken);
+        await ThrowIfAreaIsNotExistByIdAsync(command.AreaId, cancellationToken);
+        await ThrowIfDeviceIsNotExistByIdAsync(command.DeviceId, cancellationToken);
+        await ThrowIfRouterExistByNameAsync(command.Router.Name, cancellationToken);
```

Comment:
Router.Name is not unique field. Many Users can have Routers with name "Home". Otherwise, check uniqueness by User and Name is too complicated. So it would be better if Name is not unique.
Additionally, please, check if Area.Name is not unique too.